lib/pipeline.js

```diff
@@ -467,43 +476,41 @@ class PipelineModel extends BaseModel {
      * @method syncPRs
      * @return {Promise}
      */
-    syncPRs() {
-        return this.token
-            .then(token =>
-                Promise.all([
-                    this.jobs,
-                    this.scm.getOpenedPRs({
-                        scmUri: this.scmUri,
-                        scmContext: this.scmContext,
-                        token
-                    })
-                ]).then(([existingJobs, openedPRs]) => {
-                    const synced = [];
+    async syncPRs() {
+        const token = await this.token;
 
-                    openedPRs.forEach(openedPR => {
-                        const prNum = openedPR.name.split('-')[1];
+        return Promise.all([
```

Comment:
we should avoid mixing async/await & promise callbacks, these following lines could also be moved to async/await